sdk-core/src/main/java/care/data4life/sdk/migration/RecordCompatibilityService.kt

```diff
@@ -19,138 +19,101 @@ package care.data4life.sdk.migration
-                apiService.searchRecords(
-                    alias,
-                    userId,
-                    startDate,
-                    endDate,
-                    pageSize,
-                    offSet,
-                    encryptedTags.joinToString(",")
-                ),
-                BiFunction<List<EncryptedRecord>, List<EncryptedRecord>, List<EncryptedRecord>> { records1, records2 ->
-                    mutableListOf<EncryptedRecord>().also {
-                        it.addAll(records1)
-                        it.addAll(records2)
-                    }
-                }
+        tagEncryptionKey: GCKey,
+        builder: NetworkingContract.SearchTagsBuilder
+    ) {
+        tags.map { tagGroup ->
+            Pair(
+                tagGroup.key + DELIMITER,
+                compatibilityEncoder.encode(tagGroup.value)
+            )
+        }.map { encodedTagGroup ->
+            addTags(
+                encodedTagGroup.first,
+                encodedTagGroup.second,
+                tagEncryptionKey,
+                builder
+            )
+        }
+    }
+
+    private fun mapAnnotations(
+        annotations: Annotations,
+        tagEncryptionKey: GCKey,
+        builder: NetworkingContract.SearchTagsBuilder
+    ) {
+        annotations.map { annotation ->
+            Pair(
+                ANNOTATION_KEY + DELIMITER,
+                compatibilityEncoder.encode(annotation).copy(
+                    second = compatibilityEncoder.normalize(annotation)
+                )
+            )
+        }.map { encodedTagGroup ->
+            addTags(
+                encodedTagGroup.first,
+                encodedTagGroup.second,
+                tagEncryptionKey,
+                builder
             )
-        } else {
-            apiService.searchRecords(
-                alias,
-                userId,
-                startDate,
-                endDate,
-                pageSize,
-                offSet,
-                encodedAndEncryptedTags.joinToString(",")
-            ) as Observable<List<EncryptedRecord>>
         }
     }
 
-    override fun countRecords(
-        alias: String,
-        userId: String,
+    override fun resolveSearchTags(
         tags: Tags,
-        annotations: Annotations
-    ): Single<Int> {
-        val (encodedAndEncryptedTags, encryptedTags) = encrypt(tags, annotations)
+        annotation: Annotations
+    ): NetworkingContract.SearchTags {
+        val builder = searchTagsBuilderBuilderFactory.newBuilder()
+        val tagEncryptionKey = cryptoService.fetchTagEncryptionKey()
 
-        return Single.zip(
-            apiService.getCount(alias, userId, encodedAndEncryptedTags.joinToString(",")),
-            apiService.getCount(alias, userId, encryptedTags.joinToString(",")),
-            BiFunction<Int, Int, Int> { c1, c2 -> c1 + c2 }
-        )
+        mapTags(tags, tagEncryptionKey, builder)
+        mapAnnotations(annotation, tagEncryptionKey, builder)
```

Comment:
as a matter of taste, i'd like better for the builder to be configured afterwards, meaning that `mapTags` returns tags, `mapAnnotations` returns annotations, and then the builder would add the results later on. 

This way `mapTags` and `mapAnnotations` become more predictable functions without side effects and the builder gets built and sealed in one way.

Reply:
Well, then I guess I could move it all along on the top level (which I do not wanna do)...however...I can make it shorter.